src/parser/content-parser.php

```diff
@@ -146,9 +146,10 @@ public function parse( $post_content, $post_id = null, $filter_options = [] ) {
 
 		$this->warnings = [];
 
-		$has_blocks = has_blocks( $post_content );
+		// An empty `post_content` fails `has_blocks` check, but is still valid.
+		$content_is_empty = empty( trim( $post_content ) );
 
-		if ( ! $has_blocks ) {
+		if ( ! ( $content_is_empty || has_blocks( $post_content ) ) ) {
```

Comment:
Minor suggestion, the original conditional here took me a few attempts to understand

```suggestion
		$has_content = ! empty( trim( $post_content ) );
		
		if ( $has_content && ! has_blocks( $post_content ) ) {
```